src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedFiles.cs

```diff
@@ -13,120 +11,17 @@ namespace Microsoft.CodeAnalysis.Host
     internal partial class TemporaryStorageServiceFactory
     {
         /// <summary>
-        /// Rather than creating a separate MemoryMappedFile for every string/stream we need to persist,
-        /// use several large MemoryMappedFile 'arenas' that contain multiple buffers each.
+        /// Our own abstraction on top of memory map file so that we can have shared views over mmf files. 
+        /// Otherwise, each view has minimum size of 64K due to requirement forced by windows.
         /// 
-        /// Group buffers by size to simplify fragmentation and complexity of this memory management code.
```

Comment:
all these are wrong 64K minimum VM requirement is on view not map file. so this optimization actually doesnt save anything since we don't manage view but just mmf.